aws-s3-transfer-manager/tests/download_test.rs

```diff
@@ -98,6 +99,56 @@ fn simple_object_connector(data: &Bytes, part_size: usize) -> StaticReplayClient
 
     StaticReplayClient::new(events)
 }
+const OBJECT_MODIFIED_RESPONSE: &str = r#"<?xml version="1.0" encoding="UTF-8"?>
+    <Error>
+        <Code>PreconditionFailed</Code>
+        <Message>At least one of the pre-conditions you specified did not hold</Message>
+        <Condition>If-Match</Condition>
+    </Error>
+"#;
+
+/// Create a static replay client (http connector) for an object of the given size.
+///
+/// Assumptions:
+///     1. Expected requests are not created. A dummy placeholder is used. Callers need to make
+///        assertions directly on the captured requests.
+///     2. First request for discovery, succeed with etag
+///     3. Followed requests fail to mock the object changed during download.
+fn mock_object_modified_connector(data: &Bytes, part_size: usize) -> StaticReplayClient {
```

Comment:
I think we should move this to the actual test, unless it will be used in multiple tests.